internal/cli/atlas/security/ldap/save.go

```diff
@@ -0,0 +1,109 @@
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ldap
+
+import (
+	"github.com/mongodb/mongocli/internal/cli"
+	"github.com/mongodb/mongocli/internal/config"
+	"github.com/mongodb/mongocli/internal/flag"
+	"github.com/mongodb/mongocli/internal/store"
+	"github.com/mongodb/mongocli/internal/usage"
+	"github.com/spf13/cobra"
+	atlas "go.mongodb.org/atlas/mongodbatlas"
+)
+
+type SaveOpts struct {
+	cli.GlobalOpts
+	cli.OutputOpts
+	hostname              string
+	port                  int
+	bindUsername          string
+	bindPassword          string
+	caCertificate         string
+	authzQueryTemplate    string
+	authenticationEnabled bool
+	authorizationEnabled  bool
+	store                 store.LDAPConfigurationSaver
+}
+
+func (opts *SaveOpts) initStore() error {
+	var err error
+	opts.store, err = store.New(config.Default())
+	return err
+}
+
+var saveTemplate = `HOSTNAME	PORT	AUTHENTICATION	AUTHORIZATION
+{{.LDAP.Hostname}}	{{.LDAP.Port}}	{{.LDAP.AuthenticationEnabled}}	{{.LDAP.AuthorizationEnabled}}
+`
+
+func (opts *SaveOpts) Run() error {
+	r, err := opts.store.SaveLDAPConfiguration(opts.ConfigProjectID(), opts.newLDAPConfiguration())
+	if err != nil {
+		return err
+	}
+
+	return opts.Print(r)
+}
+
+func (opts *SaveOpts) newLDAPConfiguration() *atlas.LDAPConfiguration {
+	return &atlas.LDAPConfiguration{
+		LDAP: &atlas.LDAP{
+			AuthenticationEnabled: opts.authenticationEnabled,
+			AuthorizationEnabled:  opts.authorizationEnabled,
+			Hostname:              opts.hostname,
+			Port:                  opts.port,
+			BindUsername:          opts.bindUsername,
+			BindPassword:          opts.bindPassword,
+			CaCertificate:         opts.caCertificate,
+			AuthzQueryTemplate:    opts.authzQueryTemplate,
+		},
+	}
+}
+
+// mongocli atlas security ldap save --hostname hostname --port port --bindUsername bindUsername --bindPassword bindPassword --caCertificate caCertificate
+// --authzQueryTemplate authzQueryTemplate --authenticationEnabled authenticationEnabled --authorizationEnabled authorizationEnabled [--projectId projectId]
+func SaveBuilder() *cobra.Command {
+	opts := &SaveOpts{}
+	cmd := &cobra.Command{
+		Use:   "save",
+		Short: save,
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			return opts.PreRunE(
+				opts.initStore,
+				opts.InitOutput(cmd.OutOrStdout(), saveTemplate))
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return opts.Run()
+		},
+	}
+
+	cmd.Flags().StringVar(&opts.hostname, flag.Hostname, "", usage.LDAPHostname)
+	cmd.Flags().IntVar(&opts.port, flag.Port, 636, usage.LDAPPort)
```

Comment:
[q] Just to check, if you run the command with help, the default value will be shown?

Reply:
Yes it does two times now 😄 , I am going to change this xD.

```
./bin/mongocli atlas security ldap save --help
Save an LDAP configuration.

Usage:
  mongocli atlas security ldap save [flags]

Flags:
      --authenticationEnabled       Specifies whether user authentication with LDAP is enabled.
      --authorizationEnabled        Specifies whether user authorization with LDAP is enabled.
      --authzQueryTemplate string   An LDAP query template that Atlas executes to obtain the LDAP groups to which the authenticated user belongs.
      --binPassword string          The password used to authenticate the bindUsername.
      --bindUsername string         The user DN that Atlas uses to connect to the LDAP server.
      --caCertificate string        CA certificate used to verify the identify of the LDAP server.
  -h, --help                        help for save
      --hostname string             The hostname or IP address of the LDAP server.
  -o, --output string               Output format.
                                    Valid values: json
      --port int                    The port to which the LDAP server listens for client connections. Default: 636 (default 636)
      --projectId string            Project ID to use. Overrides configuration file or environment variable settings.
```

Reply:
new one

```
./bin/mongocli atlas security ldap save --help
Save an LDAP configuration.

Usage:
  mongocli atlas security ldap save [flags]

Flags:
      --authenticationEnabled       Specifies whether user authentication with LDAP is enabled.
      --authorizationEnabled        Specifies whether user authorization with LDAP is enabled.
      --authzQueryTemplate string   An LDAP query template that Atlas executes to obtain the LDAP groups to which the authenticated user belongs.
      --binPassword string          The password used to authenticate the bindUsername.
      --bindUsername string         The user DN that Atlas uses to connect to the LDAP server.
      --caCertificate string        CA certificate used to verify the identify of the LDAP server.
  -h, --help                        help for save
      --hostname string             The hostname or IP address of the LDAP server.
  -o, --output string               Output format.
                                    Valid values: json
      --port int                    The port to which the LDAP server listens for client connections. (default 636)
      --projectId string            Project ID to use. Overrides configuration file or environment variable settings.

Global Flags:
  -P, --profile string   Profile to use from your configuration file.
```

Reply:
Hehehe, nice inadvertent catch! :D